frontends/common/parser_options.cpp

```diff
@@ -355,53 +357,35 @@ void ParserOptions::setInputFile() {
     }
 }
 
-namespace {
-
-bool setIncludePathIfExists(const char *&includePathOut, const char *possiblePath) {
-    struct stat st;
-    if (!(stat(possiblePath, &st) >= 0 && S_ISDIR(st.st_mode))) return false;
-    if (auto path = realpath(possiblePath, NULL))
-        includePathOut = path;
-    else
-        includePathOut = strdup(possiblePath);
-    return true;
-}
-
-}  // namespace
-
 bool ParserOptions::searchForIncludePath(const char *&includePathOut,
-                                         std::vector<cstring> userSpecifiedPaths,
+                                         const std::vector<cstring> &userSpecifiedPaths,
                                          const char *exename) {
```

Comment:
Can `exename` be converted to `std::filesystem::path` while here?

Reply:
This would break downstream use of that function, figured I should do that in a separate PR. Ditto for all the cstring typing of variables that should be a path.